lib/src/game/viewport.dart

```diff
@@ -0,0 +1,155 @@
+import 'dart:math' as math;
+import 'dart:ui';
+
+import 'package:flutter/painting.dart';
+
+import '../../extensions.dart';
+import '../../game.dart';
+
+/// A viewport is a class that potentially translates and resizes the screen.
+/// The reason you might want to have a viewport is to make sure you handle any
+/// screen size and resolution correctly depending on your needs.
+///
+/// Not only screens can have endless configurations of width and height with
+/// different ratios, you can also embed games as widgets within a Flutter app.
+/// In fact, the size of the game can even change dynamically (if the layout
+/// changes or in desktop, for example).
+///
+/// For some simple games, that is not an issue. The game will just adapt
+/// to fit the screen, so if the game world is 1:1 with screen it will just
+/// be bigger or smaller. But if you want a consistent experience across
+/// platforms and players, you should use a viewport.
+///
+/// When using a viewport, [resize] should be called by the engine with
```

Comment:
This doc there are some references to raw size like `getRawSize` and we changed it to `canvasSize`,  I think a replace all would be good! There some variable names as well that could be renamed